tests/src/python/test_qgsfeatureiterator.py

```diff
@@ -465,10 +469,7 @@ def callback(feature):
         self.assertEqual(res, ['a'])
 
         # check with filter fids
-        res = [f['x'] for f in
```

Comment:
Why did you drop this test?

Reply:
I didn't drop it, I move it lower down to have res as a valid reference to compare the resulting fixed and skipped geometry.

Reply:
@nyalldawson satisfied with my answer on the test changes? I also now have added a feature that cannot be fixed, to trigger the abort on fail.